src/forms/FormControl.scss

```diff
@@ -0,0 +1,382 @@
+      color: var(--color-fg-muted);
+      cursor: pointer;
+      user-select: none;
+      background-color: transparent;
+      border: solid var(--primer-borderWidth-thin, 1px) transparent;
+      border-radius: var(--primer-borderRadius-medium, 6px);
+      transition: 0.2s cubic-bezier(0.3, 0, 0.5, 1);
+      transition-property: color, background-color, border-color;
+      grid-row: 1;
+      place-content: center;
+      place-self: center end;
+      grid-column: 2 / 3;
+
+      &:hover {
+        background-color: var(--color-btn-hover-bg);
+        border: solid var(--primer-borderWidth-thin, 1px) var(--color-btn-hover-bg);
+      }
+
+      &[disabled] {
+        color: var(--color-primer-fg-disabled);
+        pointer-events: none;
+      }
+    }
+
+    // sizes
+
+    // these selectors can be refactored to use :has()
+    &.FormControl--small {
+      height: var(--primer-control-small-size, 28px);
+      font-size: var(--primer-text-body-size-small, 12px);
+
+      .FormControl--input-trailingAction {
+        width: calc(var(--primer-control-small-size, 28px) - var(--base-size-8, 8px));
+        height: calc(var(--primer-control-small-size, 28px) - var(--base-size-8, 8px));
+      }
+    }
+
+    &.FormControl--medium {
+      height: var(--primer-control-medium-size, 32px);
+
+      .FormControl--input-trailingAction {
+        width: calc(var(--primer-control-medium-size, 32px) - var(--base-size-8, 8px));
+        height: calc(var(--primer-control-medium-size, 32px) - var(--base-size-8, 8px));
+      }
+    }
+
+    &.FormControl--large {
+      height: var(--primer-control-large-size, 40px);
+      padding-inline: var(--primer-control-large-paddingInline-normal, 12px);
+
+      .FormControl--input-trailingAction {
+        width: calc(var(--primer-control-large-size, 40px) - var(--base-size-8, 8px));
+        height: calc(var(--primer-control-large-size, 40px) - var(--base-size-8, 8px));
+        margin-right: calc(calc(var(--primer-control-medium-paddingInline-normal, 12px) - 0.125rem) * -1);
+      }
+    }
+  }
+}
+
+.FormControl-fieldWrap--select {
+  grid-template-columns: minmax(0, auto) var(--base-size-16, 16px);
+  padding-inline-end: var(--primer-control-medium-paddingInline-condensed, 8px);
+
+  // mask allows for background-color to respect themes
+  &::after {
+    width: var(--base-size-16, 16px);
+    height: var(--base-size-16, 16px);
+    pointer-events: none;
+    content: '';
+    background-color: var(--color-fg-muted);
+    mask: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIGhlaWdodD0iMTYiIHZpZXdCb3g9IjAgMCAxNiAxNiIgZmlsbD0iIzU4NjA2OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48cGF0aCBkPSJNNC40MjcgOS40MjdsMy4zOTYgMy4zOTZhLjI1MS4yNTEgMCAwMC4zNTQgMGwzLjM5Ni0zLjM5NkEuMjUuMjUgMCAwMDExLjM5NiA5SDQuNjA0YS4yNS4yNSAwIDAwLS4xNzcuNDI3ek00LjQyMyA2LjQ3TDcuODIgMy4wNzJhLjI1LjI1IDAgMDEuMzU0IDBMMTEuNTcgNi40N2EuMjUuMjUgMCAwMS0uMTc3LjQyN0g0LjZhLjI1LjI1IDAgMDEtLjE3Ny0uNDI3eiIgLz48L3N2Zz4=');
```

Comment:
Is browser support ok for `mask`? Not sure if it's https://caniuse.com/css-masks. autopreifixer might add `-webkit-`?